platform/windows/os_windows.cpp

```diff
@@ -384,78 +384,80 @@ LRESULT OS_Windows::WndProc(HWND hWnd, UINT uMsg, WPARAM wParam, LPARAM lParam)
+					SetCursorPos(pos.x, pos.y);
 
-				mm->set_position(c);
-				mm->set_global_position(c);
-				input->set_mouse_position(c);
-				mm->set_speed(Vector2(0, 0));
+					mm->set_position(c);
+					mm->set_global_position(c);
+					input->set_mouse_position(c);
+					mm->set_speed(Vector2(0, 0));
 
-				if (raw->data.mouse.usFlags == MOUSE_MOVE_RELATIVE) {
-					mm->set_relative(Vector2(raw->data.mouse.lLastX, raw->data.mouse.lLastY));
+					if (raw->data.mouse.usFlags == MOUSE_MOVE_RELATIVE) {
+						mm->set_relative(Vector2(raw->data.mouse.lLastX, raw->data.mouse.lLastY));
 
-				} else if (raw->data.mouse.usFlags == MOUSE_MOVE_ABSOLUTE) {
+					} else if (raw->data.mouse.usFlags == MOUSE_MOVE_ABSOLUTE) {
 
-					int nScreenWidth = GetSystemMetrics(SM_CXVIRTUALSCREEN);
-					int nScreenHeight = GetSystemMetrics(SM_CYVIRTUALSCREEN);
-					int nScreenLeft = GetSystemMetrics(SM_XVIRTUALSCREEN);
-					int nScreenTop = GetSystemMetrics(SM_YVIRTUALSCREEN);
+						int nScreenWidth = GetSystemMetrics(SM_CXVIRTUALSCREEN);
+						int nScreenHeight = GetSystemMetrics(SM_CYVIRTUALSCREEN);
+						int nScreenLeft = GetSystemMetrics(SM_XVIRTUALSCREEN);
+						int nScreenTop = GetSystemMetrics(SM_YVIRTUALSCREEN);
 
-					Vector2 abs_pos(
-							(double(raw->data.mouse.lLastX) - 65536.0 / (nScreenWidth)) * nScreenWidth / 65536.0 + nScreenLeft,
-							(double(raw->data.mouse.lLastY) - 65536.0 / (nScreenHeight)) * nScreenHeight / 65536.0 + nScreenTop);
+						Vector2 abs_pos(
+								(double(raw->data.mouse.lLastX) - 65536.0 / (nScreenWidth)) * nScreenWidth / 65536.0 + nScreenLeft,
+								(double(raw->data.mouse.lLastY) - 65536.0 / (nScreenHeight)) * nScreenHeight / 65536.0 + nScreenTop);
 
-					POINT coords; //client coords
-					coords.x = abs_pos.x;
-					coords.y = abs_pos.y;
+						POINT coords; //client coords
+						coords.x = abs_pos.x;
+						coords.y = abs_pos.y;
 
-					ScreenToClient(hWnd, &coords);
+						ScreenToClient(hWnd, &coords);
 
-					mm->set_relative(Vector2(coords.x - old_x, coords.y - old_y));
-					old_x = coords.x;
-					old_y = coords.y;
+						mm->set_relative(Vector2(coords.x - old_x, coords.y - old_y));
+						old_x = coords.x;
+						old_y = coords.y;
 
-					/*Input.mi.dx = (int)((((double)(pos.x)-nScreenLeft) * 65536) / nScreenWidth + 65536 / (nScreenWidth));
-					Input.mi.dy = (int)((((double)(pos.y)-nScreenTop) * 65536) / nScreenHeight + 65536 / (nScreenHeight));
-					*/
-				}
+						/*Input.mi.dx = (int)((((double)(pos.x)-nScreenLeft) * 65536) / nScreenWidth + 65536 / (nScreenWidth));
+						Input.mi.dy = (int)((((double)(pos.y)-nScreenTop) * 65536) / nScreenHeight + 65536 / (nScreenHeight));
+						*/
+					}
 
-				if (window_has_focus && main_loop && mm->get_relative() != Vector2())
-					input->parse_input_event(mm);
+					if (window_has_focus && main_loop && mm->get_relative() != Vector2())
+						input->parse_input_event(mm);
+				}
+				delete[] lpb;
+			} catch (const std::bad_alloc &) {
```

Comment:
We don't use exceptions in Godot's codebase. In this case, I'm not sure it provides much value, how likely is `new BYTE[dwSize]` to raise a memory allocation issue? If we get memory alloc issues here, I think there's a lot of code that will start failing in the rest of the engine anyway.

Reply:
It is actually something I noticed later, the PR dates back to January.
Since there are also conflicts with the current code, for now I would say that it can be closed. :)